src/mbest.c

```diff
@@ -124,17 +122,55 @@ void mbest_search(
 )
 {
    float   e;
-   int     i,j;
-   float   diff;
+   int     j;
 
    for(j=0; j<m; j++) {
+        float   diff;
+        int i;
+
 	e = 0.0;
 	for(i=0; i<k; i++) {
 	    diff = cb[j*k+i]-vec[i];
 	    e += diff*w[i]*diff*w[i];
 	}
+
 	index[0] = j;
-	mbest_insert(mbest, index, e);
+        if (e < mbest->list[mbest->entries - 1].error)
+	    mbest_insert(mbest, index, e);
+   }
+}
+
+/*---------------------------------------------------------------------------*\
+
+  mbest_search_equalweight
+
+  Searches vec[] to a codebbook of vectors, and maintains a list of the mbest
+  closest matches. Similar to mbest_search() but with w[] = 1.
+
+\*---------------------------------------------------------------------------*/
+
+void mbest_search_equalweight(
+		  const float  *cb,     /* VQ codebook to search         */
+		  float         vec[],  /* target vector                 */
+		  int           k,      /* dimension of vector           */
+		  int           m,      /* number on entries in codebook */
+		  struct MBEST *mbest,  /* list of closest matches       */
+		  int           index[] /* indexes that lead us here     */
+)
+{
+   float   e;
+   
+   for(int j = 0; j < m; j++) {
+        e = 0.0;
+        for(int i = 0; i < k; i++) {
+            float diff = (*cb++) - vec[i];
+            float diff2 = diff * diff;
+            e += diff2;
+        }
+
+        index[0] = j;
+        if (e < mbest->list[mbest->entries - 1].error)
+            mbest_insert(mbest, index, e);
    }
```

Comment:
@tmiw rather than add a new, parallel function - how about we just remove the weights argument `float w[]` from `mbest_search()`?  I don't think we use anything but unity weights in `newamp1.c` and `newamp2.c`.

Reply:
Works for me.

Reply:
Actually, quant_pred_mbest() uses 0 and 1, not just 1. Can we still combine them?

Reply:
Is that a function in codec2?  I can only find it in LPCNet - which has it's own set of mbest VQ functions (in it's own name space).  I forget why I did that, perhaps LPCNet wasn't linked with codec2 at the start of the project.

Reply:
Yep, codec2, in misc/vq_mbest.c